relayertest/test.go

```diff
@@ -187,15 +187,16 @@ func TestRelayer(t *testing.T, cf ibctest.ChainFactory, rf ibctest.RelayerFactor
 	)
 
 	for _, testCaseConfig := range relayerTestCaseConfigs {
-		if len(missingCapabilities(rf, testCaseConfig.RequiredRelayerCapabilities...)) > 0 {
-			continue
-		}
 		testCase := RelayerTestCase{
 			Config: testCaseConfig,
 		}
 		testCases = append(testCases, &testCase)
-		preRelayerStartFunc := func(channels []ibc.ChannelOutput) {
 
+		if len(missingCapabilities(rf, testCaseConfig.RequiredRelayerCapabilities...)) > 0 {
+			// do not execute relayer before hook if capability missing
+			continue
+		}
```

Comment:
By letting all preRelayerStartFunc's go through, it seemed to cause test pollution. So it was necessary to prevent these "before hooks" from executing. 

FWIW, it's on my radar to think more about this section of code in the near future. 

I understand it's trying to do setup once. However, it has more indirection than I'd like which makes it more difficult to determine the order of operations.

It's also setting a struct field in a closure as well as calling a nested func.